eth/protocols/snap/sync.go

```diff
@@ -1046,7 +1046,7 @@ func (s *Syncer) assignAccountTasks(success chan *accountResponse, fail chan *ac
 		// Matched a pending task to an idle peer, allocate a unique request id
 		var reqid uint64
 		for {
-			reqid = uint64(rand.Int63())
+			reqid = uint64(rand.Int64())
```

Comment:
I think we should use rand.Uint64() wherever possible instead of casting an Int64 to Uint64